### ISSUE TITLE
Three-Stage Training Schedule

### ISSUE THREAD

**Issue by MuHeDing** [EmbodiedCity/Embodied-R.code#6]
In paper, we design a three-stage training schedule.
Stage 1: In epochs 1 and 2, and 𝜔1 : 𝜔2 : 𝜔3 = 7 : 3 : 0
Stage 2: In epochs 3 and 4, and 𝜔1 : 𝜔2 : 𝜔3 = 3 : 7 : 0
Stage 3: In epochs 5 and 12, and 𝜔1 : 𝜔2 : 𝜔3 = 1 : 7 : 2

But in bash train/train_5GPUs.sh, there is only num_train_epoch=2 and reward_weights 0.7 0.1 0.2, which is inconsistent with the paper content. How can I modify it?

**Comment by serenditipy-AC** [comment]
Hi,Thank you for your interest in our project and for your keen observation. You are correct that the provided script does not match the three-stage training schedule described in the paper. The script is a simplified version for quick start.

To replicate the three-stage training, you'll need to make the following adjustments:

1.  **Save the full checkpoint:** First, you must remove the `--save_only_model true` flag from your training script. This is crucial for saving the optimizer's state, which is necessary to properly resume training between stages. Please be aware that this will require significantly more disk space.

2.  **Run training in three stages:** You will need to execute the training command three times, updating the arguments for each stage.

    ### Stage 1 (Epochs 1-2)
    This stage focuses on `choice_accuracy` and `format` with a 3:7 ratio.
    ```bash
    --model "Qwen/Qwen2.5-VL-3B-Instruct" \
    --num_train_epochs 2 \
    --reward_weights 0.3 0.7 \
    --reward_funcs choice_accuracy format \
    ```

    ### Stage 2 (Epochs 3-4)
    Load the model from Stage 1 and shift the reward weights to a 7:3 ratio.
    ```bash
    --model "path/to/your/model_from_stage_1" \
    --num_train_epochs 2 \
    --reward_weights 0.7 0.3 \
    --reward_funcs choice_accuracy format \
    ```

    ### Stage 3 (Epochs 5-12)
    Load the model from Stage 2, train for the remaining 8 epochs, and introduce the `consistency` reward. The weights are 7:1:2 for `choice_accuracy`, `format`, and `consistency` respectively.
    ```bash
    --model "path/to/your/model_from_stage_2" \
    --num_train_epochs 8 \
    --reward_weights 0.7 0.1 0.2 \
    --reward_funcs choice_accuracy format consistency \
    ```

### Alternative Recommendation

Alternatively, we have found that introducing the `consistency` reward earlier can lead to better logical consistency in the final model. For a simpler setup, you can modify the original script by just increasing the number of epochs. This trains the model with all three rewards from the beginning.

```bash
# In your original train_5GPUs.sh, simply change:
--num_train_epochs 2 \
# to:
--num_train_epochs 12 \
```

Hope this helps! Let us know if you have any other questions.

**Comment by MuHeDing** [comment]
> Hi,Thank you for your interest in our project and for your keen observation. You are correct that the provided script does not match the three-stage training schedule described in the paper. The script is a simplified version for quick start.
> 
> To replicate the three-stage training, you'll need to make the following adjustments:
> 
> 1. **Save the full checkpoint:** First, you must remove the `--save_only_model true` flag from your training script. This is crucial for saving the optimizer's state, which is necessary to properly resume training between stages. Please be aware that this will require significantly more disk space.
> 2. **Run training in three stages:** You will need to execute the training command three times, updating the arguments for each stage.
>    ### Stage 1 (Epochs 1-2)
>    This stage focuses on `choice_accuracy` and `format` with a 3:7 ratio.
>    --model "Qwen/Qwen2.5-VL-3B-Instruct" \
>    --num_train_epochs 2 \
>    --reward_weights 0.3 0.7 \
>    --reward_funcs choice_accuracy format \
>        
>          
>        
>    
>          
>        
>    
>        
>      
>    ### Stage 2 (Epochs 3-4)
>    Load the model from Stage 1 and shift the reward weights to a 7:3 ratio.
>    --model "path/to/your/model_from_stage_1" \
>    --num_train_epochs 2 \
>    --reward_weights 0.7 0.3 \
>    --reward_funcs choice_accuracy format \
>        
>          
>        
>    
>          
>        
>    
>        
>      
>    ### Stage 3 (Epochs 5-12)
>    Load the model from Stage 2, train for the remaining 8 epochs, and introduce the `consistency` reward. The weights are 7:1:2 for `choice_accuracy`, `format`, and `consistency` respectively.
>    --model "path/to/your/model_from_stage_2" \
>    --num_train_epochs 8 \
>    --reward_weights 0.7 0.1 0.2 \
>    --reward_funcs choice_accuracy format consistency \
> 
> ### Alternative Recommendation
> Alternatively, we have found that introducing the `consistency` reward earlier can lead to better logical consistency in the final model. For a simpler setup, you can modify the original script by just increasing the number of epochs. This trains the model with all three rewards from the beginning.
> 
> # In your original train_5GPUs.sh, simply change:
> --num_train_epochs 2 \
> # to:
> --num_train_epochs 12 \
> Hope this helps! Let us know if you have any other questions.

I got it ! Thank you !
